extensions-core/druid-lookups/src/main/java/io/druid/server/lookup/jdbc/JdbcDataFetcher.java

```diff
@@ -0,0 +1,202 @@
+ *  under the License.
+ * /
+ */
+
+package io.druid.server.lookup.jdbc;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.common.logger.Logger;
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.server.lookup.DataFetcher;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.util.StringMapper;
+
+import java.util.List;
+import java.util.Map;
+
+public class JdbcDataFetcher implements DataFetcher<String, String>
+{
+  private static final Logger LOGGER = new Logger(JdbcDataFetcher.class);
+
+  @JsonProperty
+  private final MetadataStorageConnectorConfig connectorConfig;
+  @JsonProperty
+  private final String table;
+  @JsonProperty
+  private final String keyColumn;
+  @JsonProperty
+  private final String valueColumn;
+
+  private final String fetchAllQuery;
+  private final String fetchQuery;
+  private final String reverseFetchQuery;
+
+  public JdbcDataFetcher(
+      @JsonProperty("connectorConfig") MetadataStorageConnectorConfig connectorConfig,
+      @JsonProperty("table") String table,
+      @JsonProperty("keyColumn") String keyColumn,
+      @JsonProperty("valueColumn") String valueColumn
+  )
+  {
+    this.connectorConfig = Preconditions.checkNotNull(connectorConfig, "connectorConfig");
+    Preconditions.checkNotNull(connectorConfig.getConnectURI(), "connectorConfig.connectURI");
+    this.table = Preconditions.checkNotNull(table, "table");
+    this.keyColumn = Preconditions.checkNotNull(keyColumn, "keyColumn");
+    this.valueColumn = Preconditions.checkNotNull(valueColumn, "valueColumn");
+
+    this.fetchAllQuery = String.format(
+        "SELECT %s, %s FROM %s",
+        this.keyColumn,
+        this.valueColumn,
+        this.table
+    );
+    this.fetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.valueColumn,
+        this.table,
+        this.keyColumn
+    );
+    this.reverseFetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.keyColumn,
+        this.table,
+        this.valueColumn
+    );
+  }
+
+  @Override
+  public Map<String, String> fetchAll()
+  {
+    final DBI dbi = getDbi();
+    return dbi.withHandle(new HandleCallback<ImmutableMap<String, String>>()
+                          {
+                            @Override
+                            public ImmutableMap<String, String> withHandle(Handle handle) throws Exception
+                            {
+                              final ImmutableMap.Builder mapBuilder = ImmutableMap.builder();
+                              List<Map<String, Object>> result = handle.select(fetchAllQuery);
+                              LOGGER.debug("Polled [%d] row(s) from table [%s]", result.size(), table);
```

Comment:
If results can stream back, I'm not sure if result.size() will be available in a guaranteed way for all connectors. Can this be changed to

```java
if(LOGGER.isDebugEnabled) {
  LOGGER.debug("Polled [%d] row(s) from table [%s]", result.size(), table);
 }
```